addons/payment_razorpay/models/payment_transaction.py

```diff
@@ -34,7 +34,8 @@ def _get_specific_processing_values(self, processing_values):
         if self.provider_code != 'razorpay':
             return res
 
-        customer_id = self._razorpay_create_customer()['id']
+        customer_is_required = self.tokenize or self.operation in ('online_token', 'offline')
+        customer_id = customer_is_required and self._razorpay_create_customer()['id']
```

Comment:
I am not sure about this one but it'll not cause any issue so LGTM
(PS: I need to check whole existence of processing values for ('online_token', 'offline') flows with @AntoineVDV )